src/Utils/ConnectionComponent.js

```diff
@@ -32,20 +33,55 @@ class ConnectionComponent extends Component {
 		return null;
 	}
 
+	static handleMockRequest(request) {
+		const mockData = RestRoutesManager.executeMockFunction(request);
+		return ConnectionComponent.buildResponse(mockData);
+	}
+
+	static cacheResponse(response) {
+		if (!response.url ||
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1067)